### PR TITLE
Refactor `create_result` for `Tuple` types to include `NamedTuple`s

### PR DESCRIPTION
**Pull request by mofeing**
@avik-pal regarding #8, does this PR fixes your problem?

### CODE REVIEW

File: src/Reactant.jl [hunk 1]
```diff
@@ -708,11 +708,11 @@ function generate_jlfunc(concrete_result, client, mod, Nargs, linear_args, linea
             push!(concrete_result_maker, :($resname = $T($(result_stores[path]))))
             return
         end
-        if T <: Tuple
+        if T <: Tuple || T <: NamedTuple
```
Comment (wsmoses) [inline src/Reactant.jl:711]: I don't think this resolves.

Specifically the resname = ... won't create a named tuple

**Comment by wsmoses** [comment]
add a test?

File: src/Reactant.jl (hunk 1 above)
Comment (mofeing) [reply src/Reactant.jl]: right, fixed.

**Comment by mofeing** [comment]
> add a test?

Added!

Why is the `CI / Julia 1.9 - macOS-latest - x64 - packaged libReactant - assertions=false - pull_request (pull_request)` workflow failing?

**Review by wsmoses — APPROVED** [review]

**Comment by wsmoses** [comment]
Seemed like random non-determinstic failure

**Pull request merged**